plugin/keyring_vault/vault_credentials_parser.h

```diff
@@ -1,42 +1,27 @@
 #ifndef MYSQL_VAULT_CREDENTIALS_PARSER_H
 #define MYSQL_VAULT_CREDENTIALS_PARSER_H
 
-#include <my_global.h>
+#include <cstddef>
+
 #include <string>
-#include <set>
-#include "vault_credentials.h"
-#include "logger.h"
 
 namespace keyring
 {
-  class Vault_credentials_parser
-  {
-  public:
-    Vault_credentials_parser(ILogger *logger)
-      : logger(logger)
-    {
-      vault_credentials_in_progress.insert(std::make_pair("vault_url", ""));
-      vault_credentials_in_progress.insert(std::make_pair("secret_mount_point", ""));
-      vault_credentials_in_progress.insert(std::make_pair("vault_ca", ""));
-      vault_credentials_in_progress.insert(std::make_pair("token", ""));
-
-      optional_value.insert("vault_ca");
-    }
-
-    bool parse(const std::string &file_url, Vault_credentials *vault_credentials);
-
-  private:
-    void reset_vault_credentials(Vault_credentials *vault_credentials);
-
-    bool parse_line(uint line_number, const Secure_string& line, Vault_credentials *vault_credentials);
-
-    bool is_valid_option(const Secure_string &option) const;
-    Vault_credentials vault_credentials_in_progress;
-    std::set<Secure_string> optional_value;
-
-    ILogger *logger;
-    static const size_t max_file_size;
-  };
+class ILogger;
+class Vault_credentials;
+
+class Vault_credentials_parser {
+ public:
+  Vault_credentials_parser(ILogger *logger) : logger_(logger) {}
+
+  bool parse(const std::string &conf_file_path,
+             Vault_credentials &vault_credentials) const;
+
+ private:
+  ILogger *logger_;
+
+  static const std::size_t max_file_size= 16384;
```

Comment:
Shall we keep such constants names in uppercase to easier distinguish them from regular variables?

Reply:
Personal preferences, STL/clib coding conventions.